libcudacxx/include/cuda/__iterator/transform_output_iterator.h

```diff
@@ -0,0 +1,500 @@
+  }
+#endif // _CCCL_STD_VER <= 2017
+
+  //! @brief Compares two \c transform_output_iterator for less than, directly comparing the stored iterators
+  _CCCL_EXEC_CHECK_DISABLE
+  template <class _Iter2 = _Iter>
+  [[nodiscard]] _CCCL_API friend constexpr auto
+  operator<(const transform_output_iterator& __lhs, const transform_output_iterator& __rhs) noexcept(
+    noexcept(_CUDA_VSTD::declval<const _Iter2&>() < _CUDA_VSTD::declval<const _Iter2&>()))
+    _CCCL_TRAILING_REQUIRES(bool)(_CUDA_VSTD::random_access_iterator<_Iter2>)
+  {
+    return __lhs.__current_ < __rhs.__current_;
+  }
+
+  //! @brief Compares two \c transform_output_iterator for greater than, directly comparing the stored iterators
+  _CCCL_EXEC_CHECK_DISABLE
+  template <class _Iter2 = _Iter>
+  [[nodiscard]] _CCCL_API friend constexpr auto
+  operator>(const transform_output_iterator& __lhs, const transform_output_iterator& __rhs) noexcept(
+    noexcept(_CUDA_VSTD::declval<const _Iter2&>() < _CUDA_VSTD::declval<const _Iter2&>()))
+    _CCCL_TRAILING_REQUIRES(bool)(_CUDA_VSTD::random_access_iterator<_Iter2>)
+  {
+    return __lhs.__current_ > __rhs.__current_;
+  }
+
+  //! @brief Compares two \c transform_output_iterator for less equal, directly comparing the stored iterators
+  _CCCL_EXEC_CHECK_DISABLE
+  template <class _Iter2 = _Iter>
+  [[nodiscard]] _CCCL_API friend constexpr auto
+  operator<=(const transform_output_iterator& __lhs, const transform_output_iterator& __rhs) noexcept(
+    noexcept(_CUDA_VSTD::declval<const _Iter2&>() < _CUDA_VSTD::declval<const _Iter2&>()))
+    _CCCL_TRAILING_REQUIRES(bool)(_CUDA_VSTD::random_access_iterator<_Iter2>)
+  {
+    return __lhs.__current_ <= __rhs.__current_;
+  }
+
+  //! @brief Compares two \c transform_output_iterator for greater equal, directly comparing the stored iterators
+  _CCCL_EXEC_CHECK_DISABLE
+  template <class _Iter2 = _Iter>
+  [[nodiscard]] _CCCL_API friend constexpr auto
+  operator>=(const transform_output_iterator& __lhs, const transform_output_iterator& __rhs) noexcept(
+    noexcept(_CUDA_VSTD::declval<const _Iter2&>() < _CUDA_VSTD::declval<const _Iter2&>()))
+    _CCCL_TRAILING_REQUIRES(bool)(_CUDA_VSTD::random_access_iterator<_Iter2>)
+  {
+    return __lhs.__current_ >= __rhs.__current_;
+  }
+
+#if _LIBCUDACXX_HAS_SPACESHIP_OPERATOR()
+  //! @brief Three-way-compares two \c transform_output_iterator, directly three-way-comparing the stored iterators
+  _CCCL_EXEC_CHECK_DISABLE
+  template <class _Iter2 = _Iter>
+  [[nodiscard]] _CCCL_API friend constexpr auto
+  operator<=>(const transform_output_iterator& __lhs, const transform_output_iterator& __rhs) noexcept(
+    noexcept(_CUDA_VSTD::declval<const _Iter2&>() <=> _CUDA_VSTD::declval<const _Iter2&>()))
+    _CCCL_TRAILING_REQUIRES(bool)(_CUDA_VSTD::random_access_iterator<_Iter2>&& _CUDA_VSTD::three_way_comparable<_Iter2>)
+  {
+    return __lhs.__current_ <=> __rhs.__current_;
+  }
+#endif // !_LIBCUDACXX_HAS_NO_SPACESHIP_OPERATOR
```

Comment:
Q: If we have `<=>`, can't we omit the other comparison operators, since they are produced via expressoon rewrites?

Reply:
A: we wont likely get `<=>` anytime soon, because that needs the orderings to be host-device annotated and they live in the host STL

Reply:
Sorry if I phrased the question unclearly. I meant: Shouldn't we write:
```c++
#if _LIBCUDACXX_HAS_SPACESHIP_OPERATOR()
  define operator<=>
#else
  define operator<
  define operator<=
  define operator>
  define operator>=
#endif
```
instead of always defining the comparison operators in the else branch?

Reply:
Discussed offline. No change requested. We are unsure whether removing the operator implementations (and falling back to spaceship) may entail regressions on older compilers.